### PR TITLE
adds schema getChanges method, updates isEqual method to accept getCh…

### PR DESCRIPTION
**Pull request by ValJed**
[PRO-5613](https://linear.app/apostrophecms/issue/PRO-5613/as-an-editor-i-can-see-how-many-changes-were-made-in-each-published)

## Summary

Adds a `getChanges` method in the `schema` module.
A wrapper calling `isEqual` with the option `getChanges`.
Instead of returning true or false, it returns an array of changed field names.

Fixes `array` field type `isEqual` method.

Adds test for new `getChanges` method.

## What are the specific steps to test this change?

Tests should pass

## What kind of change does this PR introduce?

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [ ] Related documentation has been updated
- [X] Related tests have been updated

### CODE REVIEW

**Review by boutell — DISMISSED** [review]
Nice

File: modules/@apostrophecms/schema/lib/addFieldTypes.js [hunk 1]
```diff
@@ -859,7 +859,7 @@ module.exports = (self) => {
       if (one[field.name].length !== two[field.name].length) {
         return false;
       }
-      for (let i = 0; (i < one.length); i++) {
+      for (let i = 0; (i < one[field.name].length); i++) {
```
Comment (ValJed) [inline modules/@apostrophecms/schema/lib/addFieldTypes.js:862]: Fixes isEqual method that tried to iterate over a doc object length which is null.

File: test/schemas.js [hunk 2]
```diff
@@ -2257,6 +2295,106 @@ describe('Schemas', function() {
...
+    await apos.article.insert(req, article2);
+    await apos.article.insert(req, article3);
+
+    const art1 = await apos.doc.db.findOne({ title: 'article 1' });
+    const art2 = await apos.doc.db.findOne({ title: 'article 2' });
+    const art3 = await apos.doc.db.findOne({ title: 'article 3' });
+
+    const changes11 = apos.schema.getChanges(req, apos.article.schema, art1, art1);
+    const changes12 = apos.schema.getChanges(req, apos.article.schema, art1, art2);
+    const changes23 = apos.schema.getChanges(req, apos.article.schema, art2, art3);
```
Comment (ValJed) [inline test/schemas.js:2383]: Add test for object fields too

**Review by boutell — APPROVED** [review]
approving, btw needs changelog fix

**Review by boutell — APPROVED** [review]

**Review by haroun — APPROVED** [review]

**Pull request merged**